sui_types/src/message_headers.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
+pub enum TraceTag {
+    LatencyProbe,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Header {
+    TraceId(u64),
```

Comment:
I'm not sure if you have a specific tracing protocol in mind, but we probably need much more than a u64.  If we just needed a unique ID to identify each transaction to be traced, we could simply use the Transaction digest and wouldn't need headers.  For generic tracing of any request (which is I guess the purpose of this) we should probably conform to the OpenTelemetry standard which is the emerging W3C standard for tracing.  See: 

https://www.w3.org/TR/trace-context-1/#traceparent-header

Also:

https://github.com/MystenLabs/sui/issues/852

There are other standards, like Jaeger can potentially use a 64-bit trace ID, but that would preclude us from using other tracing backends, and only Jaeger, and even Jaeger needs stuff besides the 64-bit trace ID, like a parent ID.

It's actually quite involved to do proper trace propagation, which is why I filed a ticket with specific links, it would need some investigation.

Reply:
we can certainly switch to u128 for opentelemetry - my goal here (obviously) isn't to get distributed tracing working but to start a conversation about how to handle this stuff.